mirage/fs.ml

```diff
@@ -763,18 +764,19 @@ module Make(Sectors: Mirage_block.S)(Clock : Mirage_clock.PCLOCK) = struct
               size_all t p >>= fun s -> Lwt.return @@ s + acc
           ) 0 entries
 
-    let size t key : (int, error) result Lwt.t =
+    let size t key : (Optint.Int63.t, error) result Lwt.t =
       Log.debug (fun f -> f "getting size on key %a" Mirage_kv.Key.pp key);
       match Mirage_kv.Key.segments key with
-      | [] -> size_all t root_pair >>= fun i -> Lwt.return @@ Ok i
+      | [] -> size_all t root_pair >>= fun i -> Lwt.return @@ Ok (Optint.Int63.of_int i)
       | basename::[] -> get_file_size t root_pair basename
       | segments ->
         Log.debug (fun f -> f "descending into segments %a" Fmt.(list ~sep:comma string) segments);
         Find.find_first_blockpair_of_directory t root_pair segments >>= function
-        | `Basename_on p -> size_all t p >|= fun i -> Ok i
+        | `Basename_on p -> size_all t p >|= fun i -> Ok (Optint.Int63.of_int i)
         | `No_id _ | `No_structs -> begin
             (* no directory by that name, so try for a file *)
-            Find.find_first_blockpair_of_directory t root_pair segments >>= function
+            let dirname = Mirage_kv.Key.(parent key |> segments) in 
+            Find.find_first_blockpair_of_directory t root_pair dirname >>= function
```

Comment:
why is this change needed?

Reply:
Thanks for your review!
Unfortunately I don't remember, but as my use of chamelon was for sshfs, I guess I got the file sizes wrong, but I don't have any real clues :(.
To go further, I think it'd be better to revert that change and eventually propose a PR fixing the issue later if necessary, is that ok?